Lib/collections/__init__.py

```
@@ -1151,7 +1151,10 @@ def clear(self):
         self.maps[0].clear()
 
     def __ior__(self, other):
-        self.maps[0].update(other)
+        try:
+            self.maps[0].update(other)
+        except TypeError:
+            return NotImplemented
```

Do not make a `try-except` here but rather check `isinstance` as we did for `__or__`

---

Thanks for the review!

I actually tried `isinstance(other, Mapping)` initially (to match `__or__`), but it broke existing tests (specifically `test_union_operators`).

The issue is that `__ior__` delegates to `self.maps[0].update(other)`, and `dict.update()` accepts not just Mappings but also iterables of pairs (e.g., `cm |= [('a', 1)]`). Enforcing `isinstance(other, Mapping)` would break this valid usage.

The `try-except TypeError` block seems to be the robust way to support everything `dict.update` supports (EAFP), while still correctly returning `NotImplemented` for unsupported types.

---

It's not robust. A TypeError could happen when iterating over `other` but not because it's not supported.

---

In addition, I don't think we want to support pairs as they are not supported by `|` right? It would be strange to support `cm |= [('a', 1)]` but not `cm = cm | [('a', 1)]`

---

> In addition, I don't think we want to support pairs as they are not supported by `|` right? It would be strange to support `cm |= [('a', 1)]` but not `cm = cm | [('a', 1)]`

I see your point. I agree that the current behavior creates an inconsistency where `|=` accepts iterables while `|` does not, and that `try-except` might mask other errors.

However, switching to `isinstance(other, Mapping)` effectively introduces a backward compatibility break, as it removes the support for iterable pairs that is currently enforced by the test suite (specifically the `cm3 |= pairs` case in `test_union_operators`).

If we agree that this behavior (supporting iterables in `|=`) is indeed undesirable and should be removed in favor of correctness and consistency, I can proceed with:

Updating `__ior__` to enforce `isinstance(other, Mapping)`.

Removing the conflicting test case from `Lib/test/test_collections.py`.

Please confirm if I should go ahead with this change.